### PR TITLE
Refactor price logic

### PR DESCRIPTION
**Pull request by hieronx**

### CODE REVIEW

File: test/PoolManager.t.sol [hunk 1]
```diff
@@ -795,6 +795,67 @@ contract PoolManagerTest is TestSetup {
...
+        bytes16 trancheId,
+        uint128 price
+    ) public {
+        decimals = uint8(bound(decimals, 1, 18));
+        vm.assume(poolId > 0);
+        vm.assume(trancheId > 0);
+        vm.assume(currencyId > 0);
+        centrifugeChain.addPool(poolId); // add pool
+        centrifugeChain.addTranche(poolId, trancheId, tokenName, tokenSymbol, decimals); // add tranche
+        centrifugeChain.addCurrency(currencyId, address(erc20)); // add currency
```
Comment (NunoAlexandre) [inline test/PoolManager.t.sol:814]: these comments are unecessary

File: src/PoolManager.sol [hunk 2]
```diff
@@ -55,6 +55,13 @@ struct Tranche {
     /// @dev Each tranche can have multiple liquidity pools deployed,
     ///      each linked to a unique investment currency (asset)
     mapping(address currencyAddress => address liquidityPool) liquidityPools;
+    /// @dev Each tranche has a price per liquidity pool
+    mapping(address liquidityPool => TrancheTokenPrice) prices;
+}
+
+struct TrancheTokenPrice {
```
Comment (NunoAlexandre) [inline src/PoolManager.sol:62]: 👌

**Review by NunoAlexandre — APPROVED** [review]
LGMT 👌

**Review by ilinzweilin — APPROVED** [review]

**Review by NunoAlexandre — APPROVED** [review]
✨

**Pull request merged**